Setup of development Environment in  Linux Mint 22.2

Hi @montoyatim01,
first of thank you very much for making this software and making the source code available on here. I am very impressed with the combination of simplicity and feature-richness. I am not an experienced application developer myself but I can tell that a lot of experience and good judgement went into some implementation choices. 

I have noticed quite a few minor nuisances and bugs and would like to make some modifications to the GUI and processing model and hence am currently trying to setup a development environment/build system which is turning out to be complicated and confusing. 

My system is Linux Mint 22.2 on a 64 bit platform. I installed `conan`  version 1.66.0 (via `pipx`), `cmake` version 3.28.3 and started the iterative process of running `./build.sh Release` and then dealing with the error messages that arose. Most were cmake complaining about missing libraries related to `xorg/system` and FWIW here's a list of all the packages I installed in the process. I am not positive they are all required but most were, to get through conan's initial setup.  

`libgl-dev libx11-dev  libx11-xcb-dev libgl-dev libfontenc-dev libice-dev libsm-dev libxaw7-dev libxcomposite-dev libxcursor-dev libxdamage-dev libxi-dev libxinerama-dev libxkbfile-dev libxmuu-dev libxrandr-dev libxres-dev libfontenc-devlibxaw7-devlibxkbfile-devlibxmuu-devlibxres-devlibxss-devlibxv-devlibxvmc-devlibxcb-xkb-devlibxcb-icccm4-devlibxcb-image0-devlibxcb-keysyms1-devlibxcb-render-util0-devlibxcb-xinerama0-devlibxcb-util-devlibxcb-util0-devlibxxf86vm-dev libxcb-randr0-dev libxcb-shape0libxcb-shape0-dev libxcb-sync-dev libxcb-xfixes0-dev libxcb-dri3-dev libxcb-cursor-dev libxcb-dri2-0-dev libxcb-glx0-dev libxcb-present-dev libxcb-composite0libxcb-composite0-dev libxcb-ewmh-dev libxcb-res0-dev libuuidlibuuid1 uuid` 

I'm guessing that Conan, in theory, should handle installing this stuff seeing as I set the pertinent env variable (`CONAN_SYSREQUIRES_MODE=enabled`) but I got through this bit and am only including this information for posterity. My actual question is this: I can't seem to get through the installation of `opencolorio`.  When running a "release" build, the build eventually always fails with the error: 
```  CMake Error at /home/bobbyblobby/.conan/data/openimageio/2.5.16.0/_/_/build/bb8bec3deac8fdd8830437f939fc6cdd4db3f7c4/build/Release/generators/cmakedeps_macros.cmake:39 (message):
  Library 'OpenColorIOoglapphelpers' not found in package.  If
  'OpenColorIOoglapphelpers' is a system library, declare it with
  'cpp_info.system_libs' property
Call Stack (most recent call first):
  /home/bobbyblobby/.conan/data/openimageio/2.5.16.0/_/_/build/bb8bec3deac8fdd8830437f939fc6cdd4db3f7c4/build/Release/generators/OpenColorIO-Target-release.cmake:24 (conan_package_library_targets)
  /home/bobbyblobby/.conan/data/openimageio/2.5.16.0/_/_/build/bb8bec3deac8fdd8830437f939fc6cdd4db3f7c4/build/Release/generators/OpenColorIOTargets.cmake:26 (include)
  /home/bobbyblobby/.conan/data/openimageio/2.5.16.0/_/_/build/bb8bec3deac8fdd8830437f939fc6cdd4db3f7c4/build/Release/generators/OpenColorIOConfig.cmake:16 (include)
  src/cmake/checked_find_package.cmake:132 (find_package)
  src/cmake/externalpackages.cmake:187 (checked_find_package)
  CMakeLists.txt:176 (include)

```
I'm attaching a file with the full ouput. Any pointers how to go about this would be much appreciated. 
As a last general question: Why does this project use version 1.6 of Conan? I've been reading that it's deprecated and given the fairly recent initial release of Filmvert I guess that surprised me...

[output_failed_install.txt](https://github.com/user-attachments/files/23402792/output_failed_install.txt)

--- a/conan-recipes/opencolorio/conanfile.py
+++ b/conan-recipes/opencolorio/conanfile.py
@@ -53,9 +53,8 @@ class OpenColorIOConan(ConanFile):
 
     def requirements(self):
         self.requires("expat/[>=2.6.2 <3]")
-        if self.settings.os == "Windows":
-            self.requires("glew/[>=2.1.0]")
-            self.requires("freeglut/[>=3.2.2]")
+        self.requires("glew/[>=2.1.0]")
+        self.requires("freeglut/[>=3.2.2]")
         if Version(self.version) < "2.2.0":
             self.requires("openexr/2.5.7")
         else:
@@ -146,20 +145,20 @@ class OpenColorIOConan(ConanFile):
         tc.cache_variables["CMAKE_POLICY_DEFAULT_CMP0077"] = "NEW"
         tc.cache_variables["CMAKE_POLICY_DEFAULT_CMP0091"] = "NEW"
 
+        # glew/glut
+        glew_dep = self.dependencies["glew"]
+        freeglut_dep = self.dependencies["freeglut"]
+
         if self.settings.os == "Windows":
-            # glew/glut
-            glew_dep = self.dependencies["glew"]
-            freeglut_dep = self.dependencies["freeglut"]
-
             # Convert Windows paths to use forward slashes for CMake
-            glew_root = glew_dep.package_folder.replace("\\", "/")
-            glut_root = freeglut_dep.package_folder.replace("\\", "/")
+            glew_dep = glew_dep.package_folder.replace("\\", "/")
+            freeglut_dep = freeglut_dep.package_folder.replace("\\", "/")
 
-            tc.variables["GLEW_ROOT"] = glew_root
-            tc.variables["GLUT_ROOT"] = glut_root
+            tc.variables["GLEW_ROOT"] = glew_dep
+            tc.variables["GLUT_ROOT"] = freeglut_dep
 
-            # Set specific GLEW variables that OpenColorIO expects
-            tc.variables["GLEW_INCLUDE_DIRS"] = f"{glew_root}/include"
+        # Set specific GLEW variables that OpenColorIO expects
+        tc.variables["GLEW_INCLUDE_DIRS"] = f"{glew_dep}/include"
 
         tc.generate()
 

--- a/src/main.cpp
+++ b/src/main.cpp
@@ -20,7 +20,15 @@ int main(void)
 
     // Setup Threadpool
     unsigned int numThreads = appPrefs.prefs.maxSimExports;
-    numThreads = numThreads < 1 ? 2 : numThreads;
+
+    if (numThreads == 0 || numThreads > 4096) {
+        // It's unlikely anyone has set this high, so cap it back to a sensible number
+        // This should also catch the case where std::thread::hardware_concurrency returns
+        // bad values too
+        LOG_WARN("Threads set to {}, resetting to 2", numThreads);
+        numThreads = 2;
+    }
+
     LOG_INFO("Starting thread pool with {} threads", numThreads);
     tPool = new ThreadPool(numThreads);
 

--- a/src/preferences.cpp
+++ b/src/preferences.cpp
@@ -19,8 +19,11 @@ userPreferences appPrefs;
 void userPreferences::loadFromFile() {
     try {
         std::ifstream f(getPrefFile().c_str());
-        if (!f)
+        if (!f) {
+            LOG_INFO("No preferences file found, or file handle invalid.");
             return;
+        }
+
         nlohmann::json prefObj;
         prefObj = nlohmann::json::parse(f);
         if (prefObj.contains("Preferences")) {

--- a/src/preferences.h
+++ b/src/preferences.h
@@ -38,7 +38,7 @@ struct preferenceSet {
     int debayerMode = 10;
 
     // Max simultaneous exports
-    int maxSimExports = -1;
+    int maxSimExports = 0;
 
     // OCIO
     std::string ocioPath;

